src/matrix.jl

```diff
@@ -0,0 +1,61 @@
+istable(::Type{<:Matrix}) = false
+
+rows(m::T) where {T <: Matrix} = throw(ArgumentError("a '$T' is not a table; see `?Tables.astable` for ways to treat a Matrix as a table"))
+columns(m::T) where {T <: Matrix} = throw(ArgumentError("a '$T' is not a table; see `?Tables.astable` for ways to treat a Matrix as a table"))
+
+struct MatrixTable{T}
+    names::Vector{Symbol}
+    lookup::Dict{Symbol, Int}
+    matrix::Union{Matrix{T}, Adjoint{T, Matrix{T}}}
+end
+
+istable(::Type{<:MatrixTable}) = true
+names(m::MatrixTable) = getfield(m, 1)
+
+# row interface
+struct MatrixRow{T}
+    row::Int
+    source::MatrixTable{T}
+end
+
+Base.getproperty(m::MatrixRow, ::Type{T}, col::Int, nm::Symbol) where {T} =
+    getfield(getfield(m, 2), 3)[getfield(m, 1), col]
```

Comment:
Wouldn't using symbols instead of indices be as efficient? That would be much more readable.

Reply:
This comment is still valid.